contracts/WooracleV2.sol

```diff
@@ -202,57 +217,101 @@ contract WooracleV2 is InitializableOwnable, IWooracleV2 {
-        return (0, 0);
+    function woPrice(address base) external view override returns (uint128 price, uint256 priceTimestamp) {
+        price = infos[base].price;
+        priceTimestamp = timestamp;
     }
 
     function woState(address base)
         external
         view
         override
         returns (
-            uint256 priceNow,
-            uint256 spreadNow,
-            uint256 coeffNow,
-            uint256 timestampNow
+            uint128 priceNow,
+            uint64 spreadNow,
+            uint64 coeffNow,
+            uint256 priceTimestamp
         )
     {
-        return (0, 0, 0, 0);
+        TokenInfo storage info = infos[base];
+        priceNow = info.price;
+        spreadNow = info.spread;
+        coeffNow = info.coeff;
+        priceTimestamp = timestamp;
     }
 
     function cloAddress(address base) external view override returns (address clo) {
-        clo = 0x02Bfe714e78E2Ad1bb1C2beE93eC8dc5423B66d4;
-        return clo;
+        clo = clOracles[base].oracle;
+    }
+
+    /* ----- Private Functions ----- */
+
+    function _initclOracles() private {
+        // Oracle addresses for BSC
+        address btcOracle = 0x264990fbd0A4796A3E3d8E37C4d5F87a3aCa5Ebf; // decimal: 8
+        address ethOracle = 0x9ef1B8c0E4F7dc8bF5719Ea496883DC6401d5b2e; // decimal: 8
+        address bnbOracle = 0x0567F2323251f0Aab15c8dFb1967E4e8A7D42aeE; // decimal: 8
+        address wooOracle = 0x02Bfe714e78E2Ad1bb1C2beE93eC8dc5423B66d4; // decimal: 8
+
+        address usdtOracle = 0xB97Ad0E74fa7d920791E90258A6E2085088b4320; // USDT/USD, 8
+        address usdt = 0x55d398326f99059fF775485246999027B3197955;
+
+        address wbnb = 0xbb4CdB9CBd36B01bD1cBaEBF2De08d9173bc095c;
+        address btc = 0x7130d2A12B9BCbFAe4f2634d864A1Ee1Ce3Ead9c;
+        address eth = 0x2170Ed0880ac9A755fd29B2688956BD959F933F8;
+        address woo = 0x4691937a7508860F876c9c0a2a617E7d9E945D4B;
+    }
+
+    function _setState(
+        address base,
+        uint128 newPrice,
+        uint64 newSpread,
+        uint64 newCoeff
+    ) private {
+        TokenInfo storage info = infos[base];
+        info.price = newPrice;
+        info.spread = newSpread;
+        info.coeff = newCoeff;
+    }
+
+    function _refPrice(address fromToken, address toToken)
+        private
+        view
+        returns (uint256 refPrice, uint256 refTimestamp)
+    {
+        address baseOracle = clOracles[fromToken].oracle;
+        address quoteOracle = clOracles[toToken].oracle;
+        uint8 baseDecimal = clOracles[fromToken].decimal;
+        uint8 quoteDecimal = clOracles[toToken].decimal;
+
+        (, int256 rawBaseRefPrice, , uint256 baseUpdatedAt, ) = AggregatorV3Interface(baseOracle).latestRoundData();
+        (, int256 rawQuoteRefPrice, , uint256 quoteUpdatedAt, ) = AggregatorV3Interface(quoteOracle).latestRoundData();
+        uint256 baseRefPrice = uint256(rawBaseRefPrice);
+        uint256 quoteRefPrice = uint256(rawQuoteRefPrice);
```

Comment:
No need to define these two new variables, right?

Reply:
rawBaseRefPrice is int256, these two new variables is used store uint256 type